tests/frontend/test_engine.py

```diff
@@ -166,209 +175,299 @@ def inspect():
         assert inspect() == expected2
 
 
-class TestStarshipEngine:
-    """
-    Tests various methods on the remote engine StarshipEngine.
+@pytest.fixture
+def connection():
+    """A mock connection object."""
+    return Connection(token="token", host="host", port=123, use_ssl=True)
+
+
+def mock_return(return_value):
+    """A helper function for defining a mock function that returns the given value for
+    any arguments.
     """
+    return lambda *args, **kwargs: return_value
 
-    def test_init(self, monkeypatch):
-        """
-        Tests that a StarshipEngine instance is correctly initialized when additional APIClient
-        parameters are passed.
-        """
-        mock_api_client = MagicMock()
-        monkeypatch.setattr("strawberryfields.engine.APIClient", mock_api_client)
-        engine = StarshipEngine("chip0")
-        assert engine.client == mock_api_client()
-        assert engine.jobs == []
-        assert engine.REMOTE == True
-
-    def test_reset(self, starship_engine):
-        """
-        Tests that StarshipEngine.jobs is correctly cleared when callling StarshipEngine.reset.
-        """
-        starship_engine.jobs.append(MagicMock())
-        assert len(starship_engine.jobs) == 1
-        starship_engine.reset()
-        assert len(starship_engine.jobs) == 0
 
-    def test__get_blackbird(self, starship_engine, monkeypatch):
+def mock_response(status_code, json_body=None, binary_body=None):
```

Comment:
If you would like, this could also become a `MockResponse` class and then there would be no need to use `MagicMock` (if I'm not missing something). The `__init__` could then contain the branching logic. The upside of that would be having `pytest` as the only testing framework in this file.

Reply:
Great idea. I've replaced `mock_response` with a `MockResponse` class!